Raspi/Test/MockSensor.py

```diff
@@ -1,55 +1,58 @@
-        return {
-        "x": round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals),
-        "y": round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals),
-        "z": round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals)}
-
-    def gyroscope(val=90):
-        # value in degrees for all axis stored in a dictionary
-        return {
-        "x": round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals),
-        "y": round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals),
-        "z": round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals)}
-
-    def battery(val=80):
-        # value as a percentage of the battery
-        return round(val + val*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals)
-
-    def reedSwitch(val=True):
-        # value as either true or false
-        return val
-
-    def GPS(latLngVal=20, speedVal=50):
-        # returns the latatude, longatude and the GPS speed of the bike stored in a dictionary
-        return {
-        "lat": round(latLngVal + latLngVal*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals),
-        "lng": round(latLngVal + latLngVal*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals),
-        "speed": round(speedVal + speedVal*random.uniform(-Sensors.percent_range, Sensors.percent_range), Sensors.decimals)}
-
-    def steering(val=0):
-        # value in degrees for the position of the steering stem
-        return round(val + random.uniform(-90, 90), Sensors.decimals)
-
-    def time():
-        # return the current time for the sensor as Unix Epoch time
-        return time.time()
+import random
+
+
+class MockSensor:
+    """ Base class to make a mock sensor that produces random data"""
+
+    def __init__(self, *ave_val, percent_range=0.05, decimals=2):
+        """
+        ave_val:        Either a single val that sets the average value for the
+                        sensor or and array of subvals formatted as
+                        (sub_value_name, ave_val). Where sub_value_name is the
+                        name that will be used in the outputed dictionary and
+                        the ave_val is similar to a single val.
+        percent_range:  Percent range for the randomly generated value
+        decimals:       Number of decimal places for the random generated val
+        """
+
+        self.ave_val = ave_val
+        self.percent_range = percent_range
+        self.decimals = decimals
+
+        if len(ave_val) == 1:
+            if (not isinstance(ave_val[0], int)
+               and not isinstance(ave_val[0], float)):
+                raise ValueError("""Single values must be either and int or a
```

Comment:
Should raise a `TypeError`. See https://stackoverflow.com/questions/48343387/valueerror-and-typeerror-in-python

Also change next `ValueError` below